client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -317,11 +326,9 @@ private final class PoolManager[F[_], A <: Connection[F]](
     */
   private def disposeConnection(key: RequestKey, connection: Option[A]): Unit = {
     logger.debug(s"Disposing of connection: $stats")
-    synchronized {
-      decrConnection(key)
-      connection.foreach { s =>
-        if (!s.isClosed) s.shutdown()
-      }
+    decrConnection(key)
```

Comment:
This doesn't look right. disposeConnection now lack synchronization entirely.

Reply:
@jmcardon Good catch. I will get this fixed!

Reply:
@jmcardon This should be fixed. Let me know if you have any suggestions about the changes I made.